### PR TITLE
Add README improvements.

### LINKED ISSUE #16427: README improvements
* README fixes
** Include reference to contribute guide in READEME. Some people have had a hard time finding their way from the repo to the contribution guide.
** In the "installation from source" section, refer to the actual section in the docs, instead of to the general website
**  Update "Discussion and Development" section: remove reference to scipy mailing list, but add one to pandas-dev mailing list

### PR DESCRIPTION
**Pull request by steven-cutting**
- [Y] closes #16427 
 - [Y] whatsnew entry
** Included reference to contribute guide in READEME.
** In the "installation from source" section, refer to the actual section in the docs, instead of to the general website
** Update "Discussion and Development" section: remove reference to scipy mailing list, but add one to pandas-dev mailing list

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -225,4 +225,16 @@ Python projects, questions are welcome on the scipy-user mailing
 list. Specialized discussions or design issues should take place on
```
Comment (jorisvandenbossche) [inline README.md]: can you remove the mention of scipy-user on the line above?

I would make the logic like "general questions are welcome on the pydata mailing list, specialized discussion related to pandas development on pandas-dev"

File: README.md [hunk 2]
```diff
@@ -225,4 +225,16 @@ Python projects, questions are welcome on the scipy-user mailing
...
 the PyData mailing list / Google group:
 
-https://groups.google.com/forum/#!forum/pydata
+https://mail.python.org/mailman/listinfo/pandas-dev
+
+## Contributing to pandas
+
+**[How to contribute.](http://pandas.pydata.org/pandas-docs/stable/contributing.html)**
```
Comment (jorisvandenbossche) [inline README.md]: Small comment: I would put this after the "all contributions ... welcome" line, and make it a full sentence. Something like "The best starting point is the contributing guide". or "A detailed overview on how to contribute can be found in the contributing guide"

**Review by jorisvandenbossche — COMMENTED** [review]
Thanks! 
Added some suggestions

File: README.md (hunk 2 above)
Comment (steven-cutting) [inline README.md]: Done.

File: README.md [hunk 3]
```diff
@@ -220,9 +220,29 @@ Work on ``pandas`` started at AQR (a quantitative hedge fund) in 2008 and
...
-Since pandas development is related to a number of other scientific
-Python projects, questions are welcome on the scipy-user mailing
-list. Specialized discussions or design issues should take place on
-the PyData mailing list / Google group:
+General questions and discussions can take place on the [pydata mailing list](https://groups.google.com/forum/?fromgroups#!forum/pydata).
+Specialized discussions or design issues should take place on
+the [pandas-dev mailing list](https://mail.python.org/mailman/listinfo/pandas-dev) / Google group:
 
-https://groups.google.com/forum/#!forum/pydata
+<<<<<<< HEAD
```
Comment (jorisvandenbossche) [inline README.md]: You seem to have some eft-overs from a rebase / merge conflict (although not sure where this is coming from)

(apart from that the updates look god)

Comment (steven-cutting) [inline README.md]: Crap. I'll fix it....

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by steven-cutting** [comment]
Are you going to be able to squash them on your end? I'm not sure how to do it once it's already pushed to github.

**Comment by TomAugspurger** [comment]
@steven-cutting yeah, we squash and merge? All good @jorisvandenbossche ?

File: README.md [hunk 4]
```diff
@@ -220,9 +220,17 @@ Work on ``pandas`` started at AQR (a quantitative hedge fund) in 2008 and
 has been under active development since then.
 
 ## Discussion and Development
-Since pandas development is related to a number of other scientific
-Python projects, questions are welcome on the scipy-user mailing
-list. Specialized discussions or design issues should take place on
-the PyData mailing list / Google group:
+General questions and discussions can take place on the [pydata mailing list](https://groups.google.com/forum/?fromgroups#!forum/pydata).
```
Comment (jreback) [inline README.md]: I think adding a StackOverflow link to ``[pandas]`` tag prob should be top item for general questions.

Comment (jreback) [inline README.md]: https://stackoverflow.com/questions/tagged/pandas

Comment (jorisvandenbossche) [inline README.md]: That's true, we should mention stakcverflow somewhere as well (though the section is originally about "development" dicussion, so then mailing list is more appropriate as stack overflow

StackOverflow would maybe more suit under a title "Getting help"

Comment (steven-cutting) [inline README.md]: That does sound like a good idea. I'll add it.

Comment (steven-cutting) [inline README.md]: @jorisvandenbossche 
Just saw your last comment after I submitted the latest change.
I'll move it to Getting Help.

**Comment by steven-cutting** [comment]
With the "contributing" section being added, I feel that the "Discussion and Development" section should change to the "Discussion and Getting Help" section.

**Comment by jorisvandenbossche** [comment]
@steven-cutting I made a small edit and split it just in two separate sections ("Gettting Help" and "Discussion and development")

Thanks for the PR!

**Comment by steven-cutting** [comment]
@jorisvandenbossche 
Excellent, thanks.

**Pull request merged**